browser/brave_webcompat_reporter/webcompat_reporter_dialog.cc

```diff
@@ -0,0 +1,144 @@
+/* Copyright 2019 The Brave Authors. All rights reserved.
```

Comment:
One of the feedback from @bridiver about the implementation this is based on, `browser/brave_rewards/tip_dialog.h`, is that we shouldn't have put that in `browser/` but in `components/xyz`, so I think we should probably attempt to move this to `components/brave_webcompat_reporter/browser` (and possibly the UI to `components/brave_webcompat_reporter/ui` or similar). I can't remember why we stuck with the tip_dialog in `browser/` though, so perhaps the same reason would apply to this if it's a depdency issue...

Reply:
I'll need some help sorting out the `BUILD.gn` files if we're doing this - unless there's an example I can follow somewhere

Reply:
well, we can't use `chrome/browser` from `brave/components`, but is all the stuff in `GetDialogSize` really necessary?

Reply:
`webcompat_reporter_dialog` will need to be here to use `ConstrainedWebDialogUI`, but `webcompat_reporter_service` has been moved to `brave/components`.